app/controllers/categories_controller.rb

```diff
@@ -26,9 +28,32 @@ def index
   # GET /categories/1
   # GET /categories/1.json
   def show
+    setup_stories
     redirect_to_path(categories_path) if @category.user_id != current_user.id
   end
 
+  def tagged_moments_data
+    setup_stories
+    respond_to do |format|
+      if @moments
+        format.json do
+          render json: tagged_moments_data_json
+        end
+      end
+    end
+  end
+
+  def tagged_strategies_data
+    setup_stories
+    respond_to do |format|
+      if @strategies
+        format.json do
+          render json: tagged_strategies_data_json
```

Comment:
Same here, a one-liner would do
```suggestion
          render json: tagged_strategies_data_json if @strategies
```